### DISCUSSION TITLE
Ability to blacklist or whitelist certain parsers.

### DISCUSSION THREAD

**Discussion by doZennn** [thephpleague/commonmark#648] [Q&A] [answered]
**Description**  

This would be useful if you want a more control over how content is rendered in your site.

Example use cases: Disabling embedded images and thematic breaks in comments, or only allowing bold, italics, and underlines.

**Comment by GrahamCampbell** [comment]
This is already supported. :)

**Comment by doZennn** [comment]
Is it anywhere in the docs? I can't seem to find it.

**Comment by colinodell** [comment, marked as answer]
The general gist is that you [create your own custom extension](https://commonmark.thephpleague.com/1.0/customization/extensions/) which only registers the parsers, renderers, etc. for the features you want to support.

Here's the `CommonMarkCoreExtension` which includes almost everything: https://github.com/thephpleague/commonmark/blob/master/src/Extension/CommonMarkCoreExtension.php

Here's the `InlinesOnly` extension which only includes things you'd typically find in one-line comments: https://github.com/thephpleague/commonmark/blob/master/src/Extension/InlinesOnly/InlinesOnlyExtension.php

You can use one of those as-is or create your own version to fit your own needs.

I definitely agree that the documentation could be better - PRs would be welcome! :wink:

**Comment by colinodell** [comment]
Oh, one important note - extensions can only **add** features, not remove them, so don't use `CommonMarkConverter` or `Environment::createCommonMarkEnvironment()` as your starting point because that would include everything by default.  Instead, do this:

```diff
 use League\CommonMark\CommonMarkConverter;
 use League\CommonMark\Environment;
 
 
-$environment = Environment::createCommonMarkEnvironment();
+$environment = new Environment();
+$environment->addExtension(new MyAwesomeCustomExtension());

 $converter = new CommonMarkConverter([], $environment);
 echo $converter->convertToHtml('Hello world!');
```

**Comment by doZennn** [comment]
Thank you, managed to get it working!
